packages/@ember/-internals/glimmer/tests/integration/helpers/array-test.js

```diff
@@ -0,0 +1,202 @@
+import { RenderingTest, moduleFor } from '../../utils/test-case';
+import { strip } from '../../utils/abstract-test-case';
+import { Component } from '../../utils/helpers';
+import { set } from '@ember/-internals/metal';
+
+moduleFor(
+  'Helpers test: {{array}}',
+  class extends RenderingTest {
+    ['@test returns an array']() {
+      this.render(strip`{{#with (array \"Sergio\") as |people|}}
+                          {{#each people as |personName|}}
+                            {{personName}}
+                          {{/each}}
+                        {{/with}}`);
+
+      this.assertText('Sergio');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('Sergio');
+    }
+
+    ['@test can have more than one value']() {
+      this.render(strip`{{#with (array \"Sergio\" \"Robert\") as |people|}}
+                          {{#each people as |personName|}}
+                            {{personName}},
+                          {{/each}}
+                        {{/with}}`);
+
+      this.assertText('Sergio,Robert,');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('Sergio,Robert,');
+    }
+
+    ['@test binds values when variables are used']() {
+      this.render(
+        strip`{{#with (array personOne) as |people|}}
+                {{#each people as |personName|}}
+                  {{personName}}
+                {{/each}}
+              {{/with}}`,
+        {
+          personOne: 'Tom',
+        }
+      );
+
+      this.assertText('Tom');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('Tom');
+
+      this.runTask(() => set(this.context, 'personOne', 'Yehuda'));
+
+      this.assertText('Yehuda');
+    }
+
+    ['@test binds multiple values when variables are used']() {
+      this.render(
+        strip`{{#with (array personOne personTwo) as |people|}}
+                {{#each people as |personName|}}
+                  {{personName}},
+                {{/each}}
+              {{/with}}`,
+        {
+          personOne: 'Tom',
+          personTwo: 'Yehuda',
+        }
+      );
+
+      this.assertText('Tom,Yehuda,');
+
+      this.runTask(() => this.rerender());
+
+      this.assertText('Tom,Yehuda,');
```

Comment:
Can probably replace the runTask and assertText with `this.assertStableRerender()`